pkg/shardddl/optimism/lock.go

```diff
@@ -33,6 +34,8 @@ import (
 type Lock struct {
 	mu sync.RWMutex
 
+	cli *clientv3.Client
```

Comment:
How about we let Optimist do etcd communications, and continue making lock's duty is only recording upstream status. This could be done via TrySync/DeleteColumnsByDDLs returning something to tell Optimist put/delete it

Reply:
(will refactor later)